spec.md

```diff
@@ -1586,12 +1817,196 @@ en <= Z_valid
 cover(clk, pred, en, "X equals Y when Z is valid") : optional_name
 ```
 
+## Probes
+
+Probe references are created with `probe`{.firrtl} expressions, routed through
+the design using the `define`{.firrtl} statement, read using the
+`read`{.firrtl} expression (see [@sec:reading-probe-references]), and forced
+and released with `force`{.firrtl} and `release`{.firrtl} statements.
+
+These statements are detailed below.
+
+### Define
+
+Define statements are used to route references through the design, and may be
+used wherever is most convenient in terms of available identifiers -- their
+location is not significant other than scoping, and do not have last-connect
+semantics.  Every sink-flow probe must be the target of exactly one of these
+statements.
+
+The define statement takes a sink-flow static reference target and sets
+it to the specified reference, which must either be a probe
+expression, or a static reference source.
+
+Example:
+
+```firrtl
+module Refs:
+  input clock:  Clock
+  output a : Probe<{x: UInt, y: UInt}> ; read-only ref. to wire 'p'
+  output b : RWProbe<UInt> ; force-able ref. to node 'q', inferred width.
+  output c : Probe<UInt<1>> ; read-only ref. to register 'r'
+  output d : RWProbe<UInt<2>>[4] ; vector of ref.'s to memory data in 'm'
+  output e : Probe<Clock> ; ref. to input clock port
+
+  wire p : {x: UInt, flip y : UInt}
+  define a = probe(p) ; probe is passive
+  node q = UInt<1>(0)
+  define b = rwprobe(q)
+  reg r: UInt, clock
+  define c = probe(r)
+  mem m:
+    data-type => UInt<5>
+    depth => 4
+    ; ...
+    read-under-write => undefined
+
+  define d = rwprobe(m)
+  define e = probe(clock)
+```
+
+The target is not required to be only an identifier, it may be a field within a
+bundle or other statically known sub-element of an aggregate, for example:
+
+```firrtl
+module Foo:
+  input x : UInt
+  output y : {x: UInt, p: Probe<UInt>}
+  output z : Probe<UInt>[2]
+
+  wire w : UInt
+  w <= x
+  y.x <= w
+
+  define y.p = probe(w)
+  define z[0] = probe(w)
+  define z[1] = probe(w)
+```
+
+`RWProbe`{.firrtl} references to ports are not allowed on public-facing
+modules.
+
+#### Probes and Passive Types
+
+While `Probe`{.firrtl} inner types are passive, the type of the probed
+static reference is not required to be:
+
+```firrtl
+module Foo :
+  input x : {a: UInt, flip b: UInt}
+  output y : {a: UInt, flip b: UInt}
+  output xp : Probe<{a: UInt, b: UInt}> ; passive
+
+  wire p : {a: UInt, flip b: UInt} ; p is not passive
+  define xp = probe(p)
+  p <= x
+  y <= p
+```
+
+#### Exporting References to Nested Declarations
+
+Nested declarations (see [@sec:nested-declarations]) may be exported:
+
+```firrtl
+module RefProducer :
+  input a : UInt
+  input en : UInt<1>
+  output thereg : Probe<UInt>
+
+  when en :
+    reg myreg : UInt, clk
+    myreg <= a
+    define thereg = probe(myreg)
+```
+
+### Forwarding References Upwards
```

Comment:
I think this header is stale - you could "forward references downwards" by defining an instance's input port. Perhaps "Defining References" ?

Reply:
It accurately describes the contents of the section, but some example in the "statements" portion (here, stating what `define` does/can be used for) mentioning input reference ports is important, good call thanks!  I'll tweak the header organization as well -- this should be under `Define` not a sibling to it.